### PR TITLE
Updates for 5922 - Staff can send Group messages

### PR DESCRIPTION
**Pull request by smithellis**
* Fixes privelege escalation - 1721
* Fixes bolded message text - 1726
* Removes unneeded Avatars - 1724
* Don't show a To if there is no To - 1723
* Attempts to resolve 502 issue - 1720

### CODE REVIEW

**Review by AcharayuAiyara — APPROVED** [review]

File: kitsune/messages/utils.py [hunk 1]
```diff
@@ -18,22 +19,37 @@ def send_message(to, to_group=None, text=None, sender=None):
 
     # Assuming OutboxMessage doesn't necessarily need to be created for every recipient
     if to:
+        # This is users from the To field
+        users = {user for user in to if isinstance(user, User)}
+        # This is groups from the To field
+        groups = [group for group in to if isinstance(group, Group)]
```
Comment (escattone) [inline kitsune/messages/utils.py:25]: Is there a reason why `groups` should not be a set like `users`?

File: kitsune/messages/utils.py [hunk 2]
```diff
@@ -18,22 +19,37 @@ def send_message(to, to_group=None, text=None, sender=None):
...
+        # Lets get all the data for groups that we need
+        prefetched_groups = Group.objects.filter(
+            pk__in=[group.pk for group in groups]
+        ).prefetch_related("user_set")
+        # This is who is actually getting a message
+        # - A combo of users from To field and users from groups
+        receivers = users | {user for group in prefetched_groups for user in group.user_set.all()}
+
```
Comment (escattone) [inline kitsune/messages/utils.py:33]: I think directly getting the users might be a better way to go? In other words, something like:

```python
        receivers = users | set(User.objects.filter(groups__in=groups).distinct())
```

**Review by escattone — CHANGES_REQUESTED** [review]

File: kitsune/messages/utils.py [hunk 3]
```diff
@@ -18,22 +19,37 @@ def send_message(to, to_group=None, text=None, sender=None):
...
+            # Add the groups from the To field to the message
+            outbox_message.to_group.set(groups)
+
+    for recipient in receivers:
+        inbox_message = InboxMessage.objects.create(sender=sender, to=recipient, message=text)
+        # If we had a user, and we made them an inbox message,
+        # we should also add the groups to their message as well
+        if groups:
+            inbox_message.to_group.set(groups)
+        if Setting.get_for_user(recipient, "email_private_messages"):
```
Comment (escattone) [inline kitsune/messages/utils.py:49]: This is a query for every user in the `receivers`, so could be 10's to 100's of queries. Could we make a single bulk query for this before we enter the loop? Maybe something like:

```python
set_of_user_pks_to_email_private_message = set(Setting.objects.filter(user__in=receivers, name="email_private_messages").values_list("user", flat=True))
```

and then, within the loop, check if the recipient's PK is within that set?

File: kitsune/messages/utils.py (hunk 1 above)
Comment (smithellis) [reply kitsune/messages/utils.py:25]: No, no reason at all.

File: kitsune/messages/utils.py (hunk 2 above)
Comment (smithellis) [reply kitsune/messages/utils.py:33]: I'll try this - I thought `.distinct()` would be an expensive choice, but maybe this combination makes that a wash.

File: kitsune/messages/utils.py [hunk 4]
```diff
@@ -1,39 +1,56 @@
...
+    outbox_message.to.set(users)
+
+    if groups:
+        # Add the groups from the To field to the message
+        outbox_message.to_group.set(groups)
+
+    set_of_user_pks_to_email_private_message = set(
+        Setting.objects.filter(
+            user__in=receivers, name="email_private_messages", value=True
+        ).values_list("user__pk", flat=True)
```
Comment (escattone) [inline kitsune/messages/utils.py:41]: Just for the sake of knowledge sharing, `.values_list("user", flat=True)` is equivalent to `.values_list("user__pk", flat=True)`.

**Review by escattone — APPROVED** [review]

File: kitsune/messages/utils.py (hunk 2 above)
Comment (escattone) [reply kitsune/messages/utils.py]: Good point. I was debating that as well, whether we should drop the `.distinct()` and just let the `set()` take care of it. I was thinking that the `distinct` call in Postgres might be faster, but I really don't know.

File: kitsune/messages/utils.py (hunk 4 above)
Comment (smithellis) [reply kitsune/messages/utils.py:41]: Ah - yes, I left that when troubleshooting why this was failing. It was the `value=True` bit that was missing.

File: kitsune/messages/utils.py (hunk 3 above)
Comment (escattone) [reply kitsune/messages/utils.py]: Good catch! I was wondering if the `value` had to be checked as well.

File: kitsune/messages/utils.py (hunk 4 above)
Comment (escattone) [reply kitsune/messages/utils.py:41]: Good catch on the `value=True`!

**Pull request merged**